### PR TITLE
rearrange preferences database

### PR DESCRIPTION
**Pull request by horst-p-w-neubauer**
rearrange preferences database:
* skip advanced button
* allways show editDatabaseDefaultSqlText and its label
* use complete southern dialogspace for sql edit widget
* solves #2210

### CODE REVIEW

File: src/PreferencesDialog.ui [hunk 1]
```diff
@@ -20,7 +20,7 @@
    <item>
     <widget class="QTabWidget" name="tabWidget">
      <property name="currentIndex">
-      <number>0</number>
+      <number>1</number>
```
Comment (mgrojo) [inline src/PreferencesDialog.ui:23]: This is a common mistake while using Qt Creator. It saves the current tab page according to what you have when you press save. You have to remember to save while being in the first tab, or undo this change afterwards, before the commit.

Comment (scottfurry) [reply src/PreferencesDialog.ui:23]: @mgrojo - your not on gitter so I'll ask here and related to above.
How do you feel about disconnecting from Qt Creator?
My impression is that the IDE is starting to _get in the way_ WRT code/future changes.

Comment (horst-p-w-neubauer) [reply src/PreferencesDialog.ui]: Thanks for showing - have to keep this in mind!

File: src/PreferencesDialog.ui [hunk 2]
```diff
@@ -555,159 +555,152 @@
...
          </property>
-        </widget>
+         <property name="topMargin">
+          <number>9</number>
+         </property>
+         <property name="rightMargin">
+          <number>9</number>
+         </property>
+         <property name="bottomMargin">
+          <number>9</number>
```
Comment (MKleusberg) [inline src/PreferencesDialog.ui:574]: Can you remove these four margins of size 9? We don't have them in the other layouts, so switching between the tabs looks inconsistent. Saying this also because we should avoid explicit sizes in pixels wherever it's possible because it can give unexpected results for people with HiDPI monitors.

**Comment by mgrojo** [comment]
This seems ready to merge.

**Comment by justinclift** [comment]
I'm going to start merging outstanding changes for the 3.12.x branch today or tomorrow, then push out a "release candidate v1 build".  Is this something that should be included?

**Comment by mgrojo** [comment]
It should be safe to include it. Unless it generates conflicts, it can be included.

**Comment by justinclift** [comment]
Awesome, thanks @mgrojo. :smile:
